lib/tasks/package.rake

```diff
@@ -0,0 +1,109 @@
+require 'tempfile'
+require 'English'
+
+def check_git
+  modified = `git ls-files -m --exclude-standard .`
+  if modified.empty?
+    puts 'No uncommitted changes detected.'
+  else
+    raise "Warning: uncommitted changes!\n\n#{modified}\n"
+  end
+end
+
+def build_gem(package_dir)
+  Dir.chdir '..'
+  gemfilename = "suse-connect-#{SUSE::Connect::VERSION}.gem"
+
+  `rm suse-connect-*.gem` if Dir['*.gem'].any?
+  `gem build suse-connect.gemspec`
+
+  raise 'Gem build failed.' unless $CHILD_STATUS.exitstatus.zero?
+
+  sh "cp #{gemfilename} #{package_dir}"
+  puts "Gem built and copied to #{package_dir}." if $CHILD_STATUS.exitstatus.zero?
+end
+
+def checkout_package(obs_project, package_name, package_dir)
+  Dir.chdir package_dir
+  unless Dir['.osc'].any?
+    sh 'mkdir .tmp; mv * .tmp/'
+    sh "osc co #{obs_project} #{package_name} -o ."
+    sh 'mv .tmp/* .; rm -r .tmp/'
+    puts 'Checkout successful.' if $CHILD_STATUS.exitstatus.zero?
+  end
+  `rm *suse-connect-*.gem` if Dir['*.gem'].any?
+end
+
+def check_specfile_version(obs_project, package_name, local_spec_file)
+  file = Tempfile.new('connect-spec-rake')
+  file.close
+  `osc -A 'https://api.opensuse.org' cat '#{obs_project}' '#{package_name}' '#{package_name}.spec' > #{file.path}`
+  original_version = version_from_spec(file.path)
+  new_version      = version_from_spec(local_spec_file)
+
+  if new_version == original_version
+    puts "Version in #{package_name}.spec not changed. Changing...\n"
+    change_specfile_version(local_spec_file, original_version)
+  else
+    puts "Version change to #{new_version} in #{package_name}.spec detected."
+  end
+ensure
+  file.unlink if file
+end
+
+def change_specfile_version(specfile, old_version)
+  File.write(specfile, File.open(specfile, &:read).gsub(old_version, SUSE::Connect::VERSION))
```

Comment:
That seems dangerous. If I have two lines in the specfile:

* `Version: 0.0.1`
* `Requires: rubygem-potato >= 0.0.1`

It will change both `0.0.1` matches to `0.0.2`?

Reply:
I do not have a safe solution for this using regex and `gsub`. I will remove this feature for now :(

Reply:
The safe solution would be something along the lines of `.gsub(/^Version:\s*\Q#{old_version}\E$/, SUSE::Connect::VERSION)`.

Reply:
Ruby's regex engine doesn't actually support `\Q\E`, silly Ruby.
OK, then: `.gsub(/^Version:\s*#{Regexp.escape(old_version)}$/, SUSE::Connect::VERSION)`

Reply:
In my local test scenario, that still changed identical version numbers. :/